tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/util/ProjectImportUtil.java

```diff
@@ -500,6 +500,63 @@ else if( tempRecord.isTheme() )
         return isValid;
     }
 
+    public static IStatus validateSDKPath(final String currentPath)
+    {
+        IStatus retVal = Status.OK_STATUS;
+
+        if( !org.eclipse.core.runtime.Path.EMPTY.isValidPath( currentPath ) )
+        {
+            retVal = ProjectCore.createErrorStatus( "\"" + currentPath + "\" is not a valid path." );
+        }
+        else
+        {
+            IPath osPath = org.eclipse.core.runtime.Path.fromOSString( currentPath );
+
+            if( !osPath.toFile().isAbsolute() )
+            {
+                retVal = ProjectCore.createErrorStatus( "\"" + currentPath + "\" is not an absolute path." );
```

Comment:
No, the SDK should be responsible for being able to resolve relative
paths... otherwise how do you support them if a user puts a value like this:

app.server.parent.dir = ${sdk.dir}/../path/to/some/bundle

Just try that in a plugins-sdk and then configure that sdk in eclipse
(importing aproject) and try it out how validate works.  If validate works
then that means that it was able to "resolve" the relative path.

However, in your import wizard it can't resolve that because it says that
the current property is "not an absolute path.  Just try to use this wizard
from any sdk that has a relative path in the app.server.* properties and
you will see that warning.

On Wed, Sep 16, 2015 at 2:02 AM, simon.jiang <notifications@github.com>
wrote:

> In
> tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/util/ProjectImportUtil.java
> <https://github.com/gamerson/liferay-ide/pull/1142#discussion_r39598909>:
>
> > @@ -500,6 +500,63 @@ else if( tempRecord.isTheme() )
> >          return isValid;
> >      }
> >
> > +    public static IStatus validateSDKPath(final String currentPath)
> > +    {
> > +        IStatus retVal = Status.OK_STATUS;
> > +
> > +        if( !org.eclipse.core.runtime.Path.EMPTY.isValidPath( currentPath ) )
> > +        {
> > +            retVal = ProjectCore.createErrorStatus( "\"" + currentPath + "\" is not a valid path." );
> > +        }
> > +        else
> > +        {
> > +            IPath osPath = org.eclipse.core.runtime.Path.fromOSString( currentPath );
> > +
> > +            if( !osPath.toFile().isAbsolute() )
> > +            {
> > +                retVal = ProjectCore.createErrorStatus( "\"" + currentPath + "\" is not an absolute path." );
>
> the validateSDKPath method just validate the sdk itself path and it won't
> validate the app.server.parent.dir path. The sdk validate mthod mainly
> check the sdk internal properties. Do you mean we just call the
> SDKUtil.isValidSDKLocation method to verify sdk location is validation?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gamerson/liferay-ide/pull/1142/files#r39598909>.
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com